### ISSUE TITLE
Error and warning on importing Aer

### ISSUE THREAD

**Issue by prasanthcakewalk** [Qiskit/qiskit-aer#1019]
### Informations

- **Qiskit Aer version**: 0.7.0
- **Python version**: 3.8.6
- **Operating system**: Windows 10

### What is the current behavior?

- `import qiskit` raises an warning that Aer provider could not be imported.

> C:\Users\prasanth\AppData\Roaming\Python\Python38\site-packages\qiskit\\_\_init\_\_.py:62: RuntimeWarning: Could not import the Aer provider from the qiskit-aer package. Install qiskit-aer or check your installation.
> &nbsp;&nbsp;warnings.warn('Could not import the Aer provider from the qiskit-aer '

- After importing qiskit (with the warning), trying to access `qiskit.Aer` raises an error because Aer hasn't been imported into qiskit's root namespace. This issue goes away when the package is reloaded using `importlib.reload(qiskit)`.

This is related to the issue here: https://stackoverflow.com/questions/53929599/cannot-import-aer-from-qiskit-0-7, summarized below:
- `from qiskit import Aer` fails with error.
![Screenshot 2020-11-08 132059](https://user-images.githubusercontent.com/2957516/98479648-4aac1e80-21c5-11eb-88d7-fe18c9d98d1f.png)

- But `from qiskit.providers.aer import Aer` works

### Steps to reproduce the problem

I'm not sure if this issue is reproducible on all platforms, but I'm attaching a screenshot of a minimal example demonstrating the error:
![Screenshot 2020-11-08 123430](https://user-images.githubusercontent.com/2957516/98472382-bb047100-21c0-11eb-9f7e-025a99e5ce46.png)

### What is the expected behavior?

- No warning on importing qiskit.
- `from qiskit import Aer` works without error.

### Suggested solutions

Since reloading qiskit fixed the issue, I figured changing the order of operations could fix it. I experimented with my local qiskit and found that editing this file: https://github.com/Qiskit/qiskit-terra/blob/stable/0.16/qiskit/__init__.py by
Moving lines 82-86:
```
from .version import __version__  # noqa
from .version import _get_qiskit_versions  # noqa


__qiskit_version__ = _get_qiskit_versions()
```
to before importing Aer in lines 55-64:
```
# Try to import the Aer provider if installed.
try:
    from qiskit.providers.aer import Aer
except ImportError:
    suppress_warnings = os.environ.get('QISKIT_SUPPRESS_PACKAGING_WARNINGS', '')
    if suppress_warnings.upper() != 'Y':
        if not _config.get('suppress_packaging_warnings') or suppress_warnings.upper() == 'N':
            warnings.warn('Could not import the Aer provider from the qiskit-aer '
                          'package. Install qiskit-aer or check your installation.',
                          RuntimeWarning)
```
fixes both issues (removes warning on import and makes `from qiskit import Aer` work). However, I'm not sure if this the best way to fix the issue.
 
I also tried taking the `from qiskit.providers.aer import Aer` line in the `__init__.py` file out of the try-except block to trace the cause of the error. That causes `import qiskit` to fail with an error raised at this file: https://github.com/Qiskit/qiskit-aer/blob/stable/0.7/qiskit/providers/aer/backends/qasm_simulator.py at line 25 `from .controller_wrappers import qasm_controller_execute`. Maybe this will be useful in resolving this?
![Screenshot 2020-11-08 131408](https://user-images.githubusercontent.com/2957516/98477344-5d722380-21c4-11eb-872a-40fbd20b7ab3.png)

**Comment by rudysemola** [comment]
I try to install qiskit and I have the similar (or maybe the same) problem.
When digit the qiskit.__qiskit_version__ I see this:
{'qiskit-terra': '0.16.0',
 'qiskit-aer': None,
 'qiskit-ignis': '0.5.0',
 'qiskit-ibmq-provider': '0.11.0',
 'qiskit-aqua': '0.8.0',
 'qiskit': '0.23.0'}

Why this problem for a simple installation?

**Comment by prasanthcakewalk** [comment]
Closing this, as it appears to have been fixed by the qiskit 0.23.1 update:
- qiskit-terra: 0.16.1
- qiskit-aer: 0.7.1
- qiskit-ignis: 0.5.1
- qiskit-ibmq-provider: 0.11.1
- qiskit-aqua: 0.8.1
